src/game_objects/game_object.h

```diff
@@ -24,13 +29,16 @@ class GameObject {
     btRigidBody *rigidbody;
     int id;
 public:
+    static std::unordered_map<int, GameObject*> game_objects;
     std::vector<GameObject *> children;
 
     Transform transform; // World matrix now controlled using the Transform Component
-    std::string tag; // Identify this GameObject by a human-readable tag.
+    glm::vec3 direction;
+    Tag tag; // Identify this GameObject by a tag.
```

Comment:
A bit confused as to why this changed... The original purpose was to serve as a debugging aid, no? It seems like we're using this for type checking now, in which case we can just as easily `dynamic_cast`, or even better (but more work), use runtime polymorphism.